src/main/java/com/epam/grid/engine/service/HealthCheckProviderService.java

```diff
@@ -39,43 +31,26 @@
 @Service
 public class HealthCheckProviderService {
 
-    private final EngineType engineType;
-
-    private Map<EngineType, HealthCheckProvider> providers;
+    private final HealthCheckProvider healthCheckProvider;
 
     /**
-     * This method sets grid engine type in the context.
-     * @param engineType type of grid engine
-     * @see EngineType
+     * Constructor, sets created healthCheckProvider bean to the class field.
+     * @param healthCheckProvider created HealthCheckProvider
+     * @see HealthCheckProvider
      */
-    public HealthCheckProviderService(@Value("${grid.engine.type}") final EngineType engineType) {
-        this.engineType = engineType;
+    @Autowired
```

Comment:
Can we use `@RequiredArgsConstructor` here?

Reply:
Please, fix such situation for other cases too

Reply:
done